### ISSUE TITLE
new api: +towgs84 requires proj_def.dat

### ISSUE THREAD

**Issue by c0nk** [OSGeo/PROJ#784]
Consider the following projection:
```
proj_create(ctx,
    "+proj=somerc +lat_0=46.95240555555556 +lon_0=7.439583333333333 +k_0=1 "
    "+x_0=2600000 +y_0=1200000 +ellps=bessel "
    "+towgs84=-674.374,-15.056,-405.346,0,0,0,0 +units=m +no_defs");
```
Apparently, this relies on `proj_def.dat` containing `ellps=WGS84` so it "knows" the ellps for the "cart" transformation. This seems like a bug to me. Doesn't "towgs84" imply transforming to/from WGS84? I am using proj.4 in an embedded environment where proj_def.dat doesn't exist. In this case the above code produces:
```
pj_ellipsoid - final: a=6377397.155 f=1/299.153, errno=0
pj_ellipsoid - final:    ellps=bessel
get_init: searching cache for key: [proj_def.dat:general]
get_init: searching on in init files for [proj_def.dat:general]
get_init_string: searching for section [general] in init file [proj_def.dat]
pj_open_lib(proj_def.dat): call fopen(/usr/local/share/proj/proj_def.dat) - failed
get_init: searching cache for key: [proj_def.dat:helmert]
get_init: searching on in init files for [proj_def.dat:helmert]
get_init_string: searching for section [helmert] in init file [proj_def.dat]
pj_open_lib(proj_def.dat): call fopen(/usr/local/share/proj/proj_def.dat) - failed
Helmert parameters:
x=  -674.37400  y=  -15.05600  z=  -405.34600
rx=  0.00000  ry=  0.00000  rz=  0.00000
s=   0.00000  exact=0  transpose=0
dx=  0.00000  dy=  0.00000  dz=  0.00000
drx= 0.00000  dry= 0.00000  drz= 0.00000
ds=  0.00000  t_epoch= 0.00000  t_obs= 0.00000
get_init: searching cache for key: [proj_def.dat:general]
get_init: searching on in init files for [proj_def.dat:general]
get_init_string: searching for section [general] in init file [proj_def.dat]
pj_open_lib(proj_def.dat): call fopen(/usr/local/share/proj/proj_def.dat) - failed
get_init: searching cache for key: [proj_def.dat:cart]
get_init: searching on in init files for [proj_def.dat:cart]
get_init_string: searching for section [cart] in init file [proj_def.dat]
pj_open_lib(proj_def.dat): call fopen(/usr/local/share/proj/proj_def.dat) - failed
pj_init_ctx: Must specify ellipsoid or sphere
```
I can work around the issue by applying the transformations manually:
```
proj_create(ctx,
    "+proj=pipeline "
    "+step +proj=cart +ellps=WGS84 +no_defs "
    "+step +proj=helmert x=674.37400 y=15.05600 z=405.34600 +no_defs +inv "
    "+step +proj=cart +ellps=bessel +no_defs +inv "
    "+step +proj=somerc +lat_0=46.95240555555556 +lon_0=7.439583333333333 "
    "+k_0=1 +x_0=2600000 +y_0=1200000 +ellps=bessel +units=m +no_defs ");
```

**Comment by busstoptaktik** [comment]
That's definitely an interesting corner case, which I will address as part of the wrap-up I mention in my comment over at #783.

I had noticed the bug during tonight's work, and repaired it in a compact, but slightly hacky way, which still needs access to `proj_def.dat`.

Tomorrow, I will implement a less hacky, but slightly longer, solution, that will eliminate the need for accessing `proj_def.dat`.

Your use case, by the way, is another argument for eliminating `proj_def.dat`, as argued in #201

**Comment by busstoptaktik** [comment]
Resolved in 715ec0b90aec05f15554c6591ea8cfd09d5cf043 (in connection with a number of earlier commits)
